homeassistant/components/pyload/sensor.py

```diff
@@ -100,33 +109,42 @@ async def async_setup_platform(
             f"Authentication failed for {config[CONF_USERNAME]}, check your login credentials"
         ) from e
 
-    devices = []
-    for ng_type in monitored_types:
-        new_sensor = PyLoadSensor(
-            api=pyloadapi, sensor_type=SENSOR_TYPES[ng_type], client_name=name
-        )
-        devices.append(new_sensor)
-
-    async_add_entities(devices, True)
+    async_add_entities(
+        (
+            PyLoadSensor(
+                api=pyloadapi, entity_description=description, client_name=name
+            )
+            for description in SENSOR_DESCRIPTIONS
+        ),
+        True,
+    )
 
 
 class PyLoadSensor(SensorEntity):
     """Representation of a pyLoad sensor."""
 
     def __init__(
-        self, api: PyLoadAPI, sensor_type: SensorEntityDescription, client_name
+        self, api: PyLoadAPI, entity_description: SensorEntityDescription, client_name
     ) -> None:
         """Initialize a new pyLoad sensor."""
-        self._attr_name = f"{client_name} {sensor_type.name}"
-        self.type = sensor_type.key
+        self._attr_name = f"{client_name} {entity_description.name}"
+        self.type = entity_description.key
         self.api = api
-        self.entity_description = sensor_type
-        self.data: StatusServerResponse
+        self.entity_description = entity_description
+        self._attr_available = False
+        self.data: dict[str, Any] = {}
 
     async def async_update(self) -> None:
         """Update state of sensor."""
         try:
-            self.data = await self.api.get_status()
+            start = monotonic()
+            status = await self.api.get_status()
+            self.data = status.to_dict()
+
+            _LOGGER.debug(
+                "Finished fetching pyload data in %.3f seconds",
+                monotonic() - start,
+            )
```

Comment:
Limit the try catch scope to the command that can throw, and move the other command outside of the try/catch.

Reply:
It's done 👍